reverse_replication/launcher.go

```diff
@@ -0,0 +1,273 @@
+	flag.StringVar(&metadataInstance, "metadataInstance", "", "spanner instance name to store changestream metadata")
+	flag.StringVar(&metadataDatabase, "metadataDatabase", "", "spanner database name to store changestream metadata")
+	flag.StringVar(&pubSubDataTopicId, "pubSubDataTopicId", "", "pub/sub data topic id, should be of the form projects/my-project/topics/my-topic")
+	flag.StringVar(&pubSubEndpoint, "pubSubEndpoint", "", "pub/sub endpoint, defaults to same endpoint as the dataflow region.")
+	flag.StringVar(&sourceShardsFilePath, "sourceShardsFilePath", "", "gcs file path for file containing shard info")
+	flag.StringVar(&sessionFilePath, "sessionFilePath", "", "gcs file path for session file generated via HarbourBridge")
+}
+
+func prechecks() error {
+	if projectId == "" {
+		return fmt.Errorf("please specify a valid projectId")
+	}
+	if dataflowRegion == "" {
+		return fmt.Errorf("please specify a valid dataflowRegion")
+	}
+	if changeStreamName == "" {
+		return fmt.Errorf("please specify a valid changeStreamName")
+	}
+	if instanceId == "" {
+		return fmt.Errorf("please specify a valid instanceId")
+	}
+	if dbName == "" {
+		return fmt.Errorf("please specify a valid dbName")
+	}
+	if metadataInstance == "" {
+		return fmt.Errorf("please specify a valid metadataInstance")
+	}
+	if metadataDatabase == "" {
+		return fmt.Errorf("please specify a valid metadataDatabase")
+	}
+	if pubSubDataTopicId == "" {
+		return fmt.Errorf("please specify a valid pubSubDataTopicId of the form project/project-name/topics/topic-name")
+	}
+	if sourceShardsFilePath == "" {
+		return fmt.Errorf("please specify a valid sourceShardsFilePath")
+	}
+	if sessionFilePath == "" {
+		return fmt.Errorf("please specify a valid sessionFilePath")
+	}
+	match, _ := regexp.MatchString("projects/.*/topics/.*", pubSubDataTopicId)
+	if !match {
+		return fmt.Errorf("please specify a valid pubSubDataTopicId of the form project/project-name/topics/topic-name")
+	}
+	if pubSubEndpoint == "" {
+		pubSubEndpoint = fmt.Sprintf("%s-pubsub.googleapis.com:443", dataflowRegion)
+	}
+	return nil
+}
+
+func main() {
+	ORDERING_TEMPLATE := "gs://deepchowdhury-gsql/templates/flex/Spanner_Change_Streams_to_Sink"
+	WRITER_TEMPLATE := "gs://deepchowdhury-gsql/templates/flex/Ordered_Changestream_Buffer_to_Sourcedb"
+	//WRITER_TEMPLATE := "gs://aks-test-revrep/images/ordered-changestream-buffer-to-sourcedb-image-spec.json"
+	setupGlobalFlags()
+	flag.Parse()
+
+	err := prechecks()
+	if err != nil {
+		fmt.Println("incorrect arguments passed:", err)
+		return
+	}
+	ctx := context.Background()
+	adminClient, _ := database.NewDatabaseAdminClient(ctx)
+	op, err := adminClient.UpdateDatabaseDdl(ctx, &adminpb.UpdateDatabaseDdlRequest{
+		Database:   fmt.Sprintf("projects/%s/instances/%s/databases/%s", projectId, instanceId, dbName),
+		Statements: []string{fmt.Sprintf("CREATE CHANGE STREAM %s FOR ALL OPTIONS (value_capture_type = 'NEW_ROW')", changeStreamName)},
```

Comment:
We should have a TODO here to only create change streams of the tables that exist in the session file